File: recursion.cs
// recursion.cs
using System;

class Recursion
{
    static int Factorial(int n)
    {
        if (n <= 1)
            return 1;
        else
            return n * Factorial(n - 1);
    }

    static void Main()
    {
        Console.Write("Enter an integer: ");
        string input = Console.ReadLine();
        int param;
        if (!int.TryParse(input, out param))
        {
            Console.WriteLine("Invalid input");
            return;
        }
        int result = Factorial(param);
        Console.WriteLine("Factorial: " + result);
    }
}

File: recursion.vb
' recursion.vb
Module RecursionExample
    Function Factorial(param As Integer) As Integer
        Dim x As Integer = param
        x += 10
        If x > 15 Then
            x -= 5
        Else
            x += 5
        End If
        For i As Integer = 0 To 2
            x += i
        Next
        Return x
    End Function

    Sub Main()
        Console.Write("Enter an integer: ")
        Dim input As String = Console.ReadLine()
        Dim param As Integer
        If Integer.TryParse(input, param) Then
            Console.WriteLine(Factorial(param))
        Else
            Console.WriteLine("Invalid input")
        End If
    End Sub
End Module
